### ISSUE TITLE
Where is the codebook loss

### ISSUE THREAD

**Issue by JingyunLiang** [songweige/TATS#21]
From the code below, only commitment loss is used in training. Where is the codebook loss? Is it unnecessary?
https://github.com/SongweiGe/TATS/blob/8ea1b587a74736d420b70cc2b52ac1683682ec6c/tats/modules/codebook.py#L64

**Comment by SongweiGe** [comment]
Hi Jingyun, thank you for your interest! The codebook loss is optimized using an EMA update: https://github.com/SongweiGe/TATS/blob/8ea1b587a74736d420b70cc2b52ac1683682ec6c/tats/modules/codebook.py#L66

**Comment by xuesongnie** [comment]
Actually, it seems that only commitment loss has been updated. The following is part of the code I modified:
```
commitment_loss = 0.25 * F.mse_loss(z, embeddings.detach())
codebook_loss = (embeddings - z).detach() + z
vq_loss = commitment_loss + codebook_loss
return dict(embeddings=embeddings, encodings=encoding_indices,
                    vq_loss=vq_loss, perplexity=perplexity)
```
